File: Data Structures & Algorithms/two-integer-sum/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums: number[], target: number): number[] {
        let lookup = new Map<number, number>();

        for (let i = 0; i < nums.length; i++) {
            let complement: number = target - nums[i];

            if (lookup.has(complement)) {
                return [lookup.get(complement)!, i];
            }

            lookup.set(nums[i], i);
        }

        return [-1, -1];
    }
}

File: Data Structures & Algorithms/custom-sort-string/submission-1.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        vector<int>lookup(26, 26);

        for (int i=0; i<order.length(); i++) {
            const char ch = order[i];
            lookup[ch - 'a'] = i;
        }

        sort(s.begin(), s.end(), [lookup](char a, char b){
            return lookup[a - 'a'] < lookup[b - 'a'];
        });

        return s;
    }
};